### PR TITLE
Integration Test - Remove Support for Helm v2 and Provide Support for Enable/Disable wait while Installing Charts

### PR DESCRIPTION
**Pull request by r4rajat**
## Change Overview

Remove Support for Helm v2 and Provide Support for Enable/Disable --wait while Installing Charts

- Add wait as an argument in Install Method present in pkg/helm/client.go and pkg/helm/helm.go
- Add wait=true in apps requiring wait flag while installation 

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

```bash
$ make integration-test
```
```
running CMD in the containerized build environment
Running integration tests:
/go/src/github.com/kanisterio/kanister/pkg/testing /go/src/github.com/kanisterio/kanister
go: downloading github.com/pkg/errors v0.9.1
go: downloading k8s.io/api v0.24.3
go: downloading github.com/kastenhq/check v0.0.0-20180626002341-0264cfcea734
go: downloading k8s.io/client-go v0.24.3
go: downloading github.com/jpillora/backoff v1.0.0
go: downloading k8s.io/apimachinery v0.24.3
..
..
..
OK: 6 passed
--- PASS: Test (167.00s)
PASS
ok  	github.com/kanisterio/kanister/pkg/testing	167.037s
/go/src/github.com/kanisterio/kanister
```

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/helm/helm.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ type Client interface {
 	// RemoveRepo removes helm repo
 	RemoveRepo(ctx context.Context, name string) error
 	// Install installs helm chart with given release name in the namespace
```
Comment (viveksinghggits) [inline pkg/helm/helm.go:28]: let's improve this comment to specify the significance of `wait` as well.

**Comment by r4rajat** [comment]
ok-to-test

**Review by akankshakumari393 — COMMENTED** [review]
LGTM

File: pkg/helm/client.go [hunk 2]
```diff
@@ -125,18 +125,18 @@ func (h CliClient) UpdateRepo(ctx context.Context) error {
...
 		setVals += fmt.Sprintf("%s=%s,", k, v)
 	}
 
 	var cmd []string
-	if h.version == V3 {
+	if wait {
 		cmd = []string{"install", release, "--version", version, "--namespace", namespace, chart, "--set", setVals, "--wait", "--create-namespace"}
 	} else {
-		cmd = []string{"install", "--name", release, "--version", version, "--namespace", namespace, chart, "--set", setVals, "--wait"}
+		cmd = []string{"install", release, "--version", version, "--namespace", namespace, chart, "--set", setVals, "--create-namespace"}
```
Comment (ihcsim) [inline pkg/helm/client.go:139]: TIOLI: replace the `if/else` with an `if`:

```diff
diff --git a/pkg/helm/client.go b/pkg/helm/client.go
index 4103a26d..159827cf 100644
--- a/pkg/helm/client.go
+++ b/pkg/helm/client.go
@@ -132,11 +132,9 @@ func (h CliClient) Install(ctx context.Context, chart, version, release, namespa
 		setVals += fmt.Sprintf("%s=%s,", k, v)
 	}
 
-	var cmd []string
+	cmd := []string{"install", release, "--version", version, "--namespace", namespace, chart, "--set", setVals, "--create-namespace"}
 	if wait {
-		cmd = []string{"install", release, "--version", version, "--namespace", namespace, chart, "--set", setVals, "--wait", "--create-namespace"}
-	} else {
-		cmd = []string{"install", release, "--version", version, "--namespace", namespace, chart, "--set", setVals, "--create-namespace"}
+		cmd = append(cmd, "--wait")
 	}
 
 	out, err := RunCmdWithTimeout(ctx, h.helmBin, cmd)
```

**Review by ihcsim — APPROVED** [review]
LGTM - thanks for working on this :+1:. 

I left a TIOLI comment at `pkg/helm/client.go`. Feel free to take it or leave it. 

Also, I think all the apps should be using `NewHelmApp()` instead of calling the `NewCliClient()` directly. But that can be a refactoring for another day.

File: pkg/helm/client.go (hunk 2 above)
Comment (r4rajat) [reply pkg/helm/client.go]: Looks Good !!

**Pull request merged**